### ISSUE TITLE
واژه نامه پرازش زبان طبیعی (NLP)

### ISSUE THREAD

**Issue by AlisterTA** [erfannoury/persian-ml-glossary#4]
<div dir="rtl">
ابتدای امسال کورس CS224n استنفورد را گذراندم و یک سری یاداشت برداری برای خودم کردم.
تصمیم گرفتم مرتبش کنم و به صورت یک کتابچه دربیارم. البته هنوز وقت نکردم تکمیلش کنم. یک سری واژگان وجود دارد که سعی کردم براشون معادل سازی کنم و برای برخی از اونها هنوز کلمه جالبی به ذهنم نرسیده. 
لطفا دوستان نظر خودشون رو بنویسند
</div>

Local invariance | ناوردایی   محلی |  
compositionality | ترکیب   پذیری |  
regular grids | مشبک‌های باقاعده |  
feedforward neural networks | شبکه‌های   عصبی پیش‌خور |  
local receptive fields | نواحی ارداکی محلی | نواحی تاثیر محلی
absolute positions | موقعیت   های‌ قطعی |  
natural language processing | پردازش زبان‌ طبیعی |  
n-gram | ان-گرم |  
long-range dependencies | وابستگی‌های برد-بلند |  
long-distance dependence | وابستگی‌   راه-‌دور |  
pooling | تجمیع | ادغام
word embedding vectors | بردارهای   تعبیه کلمه |  
padding | گسترش مرز |  
pre-trained | ازپیش‌آموزش‌داده‌شده |  
spatial dimension | بعد مکانی |  
latent space | فضای   پنهان |  
filter | فیلتر | پالایه
regions | نواحی |  
receptive fields | میدان‌های تاثیر |  
ReLU | یکسوساز |  
tanh | تانژانت هذلودی |  
feature map | نگاشت   ویژگی |  
bias | سوءگیری | پیشقدر
overfitting | بیش‌برارزش |  
Pooling layer | لایه تجمیع | لایه ادغام
robustness | توانمندی |  
global k-max pooling | تجمیع ک-حداکثر عمومی | ادغام ک-حداکثر عمومی
max pooling | تجمیع   حداکثر | ادغام   حداکثر
Average pooling | تجمیع میانگین | ادغام میانگین
Softmax | بیشینه‌هموار |  
sigmoid | سیگموئید |  
zero-padding | گسترش‌مرز-صفر |  
Fully connected | تماما متصل |  
Predictive regions | نواحی   متضمن پیشگویی |  
Saliency maps | نگاشتهای برجستگی |  
Recurrent Neural Networks | شبکه   های عصبی بازگشتی |  
sequences | دنباله ها |  
vanishing and exploding gradients | ناپدید   شدن و انفجار گرادیان |  
Multi-Layer Perceptron | پرسپترون چند لایه |  
Cell state | حالت   سلول |  
Explicit memory | حافظه آشکار |  
Reset gate | گیت   بازنشانی |  
Update gate | گیت بروزرسانی |  
encoder-decoder | رمزگذار-رمزگشا |  
Attention mechanism |   |  
Neural Machine Translation | ترجمه   ماشینی عصبی |  
sequence-to-sequence | دنباله به دنباله |  
end-to-end | یکپارچه |  
Encoder | رمزگذار |  
bidirectional | دوطرفه |  
Decoder | رمزگشا |  
Context vector | بردار   متن |  
concatenation | الحاق |  
locally | محلی | موضعی
Globally | سراسری |  
local   attention |   |  
Global attention |   |  
alignment vector | بردار   هم راستایی |  
Comparison function | تابع مقایسه |  
Dot product | ضرب   نقطه ای |  
truncated Normal distribution | توزیع نرمال بریده شده |  
dense | تمالا   متصل |  
Hierarchical attention |   |  
Annotation (annotation vector) |   |  
monotonic alignment |  هم راستایی یکنوا |  
predictive   alignment |   هم راستایی پیش‌گو|  
self-attention |   |

**Comment by MohsenFayyaz89** [comment]
Global attention | نگرش سراسری
Local attention | نگرش محلی
self-attention | خود نگرش

**Comment by AlisterTA** [comment]
~~رمزگذار~~
Encoder| رمزنگار

~~dense | تمالا متصل |~~
متراکم

**Comment by erfannoury** [comment]
<div dir="rtl">
@MohsenFayyaz89 نمی‌دونم چقدر ترجمه‌ی نگرش ترجمه‌ی مناسبی باشه، ولی با توجه به اهمیت بالای این کلمه به نظرم خوبه که یه Issue جدا براش بسازیم تا اونجا بتونیم در موردش صحبت کنیم.

**Comment by erfannoury** [comment]
<div dir="rtl">
@AlisterTA می‌تونی برای بعضی از کلمات مهم (که توی عبارات متعددی ازشون استفاده میشه) یه Issue جدا بسازی؟ مثلاً بجز Attention فکر می‌کنم Pooling هم کلمه‌ی مهمی هست که تو عبارات زیادی ازش استفاده میشه. من خودم از ترجمه‌ی «ادغام» تا به حال براش استفاده کردم.

**Comment by AlisterTA** [comment]
> @MohsenFayyaz89 نمی‌دونم چقدر ترجمه‌ی نگرش ترجمه‌ی مناسبی باشه، ولی با توجه به اهمیت بالای این کلمه به نظرم خوبه که یه Issue جدا براش بسازیم تا اونجا بتونیم در موردش صحبت کنیم.

<div dir="rtl">
کلمه نگرش به attitude نزدیکتره 
در نورو ساینس این کلمه به "توجه" معنی شده اما من زیاد "توجه" رو مناسب نمیدونم 
کاش دیگران هم نظر میدادند.
</div>
[Attention](https://en.wikipedia.org/wiki/Attention)

**Comment by AlisterTA** [comment]
> @AlisterTA می‌تونی برای بعضی از کلمات مهم (که توی عبارات متعددی ازشون استفاده میشه) یه Issue جدا بسازی. مثلاً بجز Attention فکر می‌کنم Pooling هم کلمه‌ی مهمی هست که تو عبارات زیادی ازش استفاده میشه. من خودم از ترجمه‌ی «ادغام» تا به حال براش استفاده کردم.

<div dir="rtl">
درسته، کلمه دیگری که من سعی دارم معادل خوبی براش بسازم کلمه Annotation  هست. در مقالات به حالات پنهان encoder (در معماری seq2seq در حالتی که مکانیسم attention بکاربرده میشه) annotations گفته میشه. چه کلمه ای پیشنهاد میدید؟
</div>

**Comment by erfannoury** [comment]
<div dir="rtl">
@AlisterTA @MohsenFayyaz89

 #5 رو برای کلمه‌ی Attention درست کردم. اونجا می‌تونیم در موردش صحبت کنیم.

**Comment by erfannoury** [comment]
<div dir="rtl">

> درسته، کلمه دیگری که من سعی دارم معادل خوبی براش بسازم کلمه Annotation هست. در مقالات به حالات پنهان encoder (در معماری seq2seq در حالتی که مکانیسم attention بکاربرده میشه) annotations گفته میشه. چه کلمه ای پیشنهاد میدید؟ 

@AlisterTA می‌تونی نمونه‌ای بیاری؟ من تا به حال استفاده از اون کلمه رو در اون کاربرد ندیدم.

**Comment by AlisterTA** [comment]
> > درسته، کلمه دیگری که من سعی دارم معادل خوبی براش بسازم کلمه Annotation هست. در مقالات به حالات پنهان encoder (در معماری seq2seq در حالتی که مکانیسم attention بکاربرده میشه) annotations گفته میشه. چه کلمه ای پیشنهاد میدید؟
> 
> @AlisterTA می‌تونی نمونه‌ای بیاری؟ من تا به حال استفاده از اون کلمه رو در اون کاربرد ندیدم.



<div dir="rtl">
به عنوان نمونه
</div>

![image](https://user-images.githubusercontent.com/19950298/48149587-b9ea9180-e2d1-11e8-8d88-1a726c8bbb05.png)

[paper](https://arxiv.org/pdf/1611.04798.pdf)

**Comment by erfannoury** [comment]
<div dir="rtl">

@AlisterTA جاهای دیگه‌ای هم استفاده میشه یا فقط این مقاله واسه خودش برداشته یه اسمی گذاشته؟

**Comment by AlisterTA** [comment]
> @AlisterTA جاهای دیگه‌ای هم استفاده میشه یا فقط این مقاله واسه خودش برداشته یه اسمی گذاشته؟
<div dir="rtl">
نه واژه ای هست که چندین بار به اون برخوردم، در مقالات مختلف

حتی Jason Brownlee در وبلاگ خودش به این موضوع اقرار میکنه
</div>

![image](https://user-images.githubusercontent.com/19950298/48151757-5bc0ad00-e2d7-11e8-8a9a-b5129c80e8f1.png)

[Jason Brownlee](https://machinelearningmastery.com/how-does-attention-work-in-encoder-decoder-recurrent-neural-networks/)

**Comment by erfannoury** [comment]
<div dir="rtl">
@AlisterTA این به نظر میاد که یه واژه‌ی قراردادی هست، یعنی قرارداد شده که در این بحث و با این معنا به کار بره. شاید بهتر باشه در این شرایط ترجمه‌ای با توجه به کاربرد و معنای محلی کلمه انتخاب بشه، نه لزوماً ترجمه‌ی مستقیم اون کلمه.

**Comment by MohsenFayyaz89** [comment]
> @AlisterTA این به نظر میاد که یه واژه‌ی قراردادی هست، یعنی قرارداد شده که در این بحث و با این معنا به کار بره. شاید بهتر باشه در این شرایط ترجمه‌ای با توجه به کاربرد و معنای محلی کلمه انتخاب بشه، نه لزوماً ترجمه‌ی مستقیم اون کلمه.

<div dir='rtl'>
منم موافقم این یه بحث قراردادی هست. من لزوما همه جا ندیدم این اصطلاح به کار برده بشه
</div>

**Comment by AlisterTA** [comment]
ممنونم از کمکتون 
اینجا منتشر کردم 
https://alisterta.github.io/2018-11-08/کتابچه-راهنمای-پردازش-زبان-طبیعی/
